cachi2/core/package_managers/yarn_classic/project.py

```diff
@@ -0,0 +1,173 @@
+        return self._data
+
+    @property
+    def path(self) -> RootedPath:
+        return self._path
+
+    @property
+    @abstractmethod
+    def config_kind(self) -> ConfigKind:
+        """Return kind of ConfigFile instance."""
+        pass
+
+    @classmethod
+    @abstractmethod
+    def from_file(cls, path: RootedPath) -> "_CommonConfigFile":
+        """Construct a ConfigFile instance."""
+        pass
+
+
+class PackageJson(_CommonConfigFile):
+    """A package.json file.
+
+    This class abstracts the underlying attributes and only exposes what
+    is relevant for the request processing.
+    """
+
+    @property
+    def config_kind(self) -> ConfigKind:
+        """Return kind of this ConfigFile."""
+        return "package_json"
+
+    @property
+    def install_config(self) -> dict[str, Any]:
+        """Get the installConfig dict."""
+        return self.data.get("installConfig", {})
+
+    @classmethod
+    def from_file(cls, path: RootedPath) -> "PackageJson":
+        """Construct a PackageJson instance."""
+        try:
+            package_json_data = json.loads(path.path.read_text())
+        except FileNotFoundError:
+            raise PackageRejected(
+                reason="The package.json file must be present for the yarn package manager",
+                solution=(
+                    "Please double-check that you have specified the correct path "
+                    "to the package directory containing this file"
+                ),
+            )
+        except json.decoder.JSONDecodeError as e:
+            raise PackageRejected(
+                reason=f"Can't parse the {path.subpath_from_root} file. {e}",
+                solution=(
+                    "The package.json file must contain valid JSON. "
+                    "Refer to the parser error and fix the contents of the file."
+                ),
+            )
+
+        return cls(path, package_json_data)
+
+
+class YarnLock(_CommonConfigFile):
+    """A yarn.lock file.
+
+    This class abstracts the underlying attributes.
+    """
+
+    yarn_lockfile: lockfile.Lockfile
+
+    @property
+    def config_kind(self) -> ConfigKind:
+        """Return kind of this ConfigFile."""
+        return "yarnlock"
+
+    @classmethod
+    def from_file(cls, path: RootedPath) -> "YarnLock":
+        """Parse the content of a yarn.lock file."""
+        try:
+            yarn_lockfile = lockfile.Lockfile.from_file(path)
+        except FileNotFoundError:
+            raise PackageRejected(
+                reason="The yarn.lock file must be present for the yarn package manager",
+                solution=(
+                    "Please double-check that you have specified the correct path "
+                    "to the package directory containing this file"
+                ),
+            )
+        except ValueError as e:
+            raise PackageRejected(
+                reason=(f"Can't parse the {path.subpath_from_root} file.\n" f"{e}"),
```

Comment:
I think giving the user a full path to the file is more valuable.

Reply:
Fixed